sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/ScalaReflectionSuite.scala

```diff
@@ -362,4 +362,38 @@ class ScalaReflectionSuite extends SparkFunSuite {
     assert(numberOfCheckedArguments(deserializerFor[(java.lang.Double, Int)]) == 1)
     assert(numberOfCheckedArguments(deserializerFor[(java.lang.Integer, java.lang.Integer)]) == 0)
   }
+
+  test("SPARK-24762: serializer for Option of Product") {
```

Comment:
do we need to add tests here? `ScalaReflection` is not updated in this PR.

Reply:
It is mainly because I don't find related tests for serializer and deserializer of Option[Product]. I can remove it or make a separate PR for it, since it is not related to the change in this PR.